go/service/main.go

```diff
@@ -639,6 +639,23 @@ func (d *Service) hourlyChecks() {
 	}()
 }
 
+func (d *Service) deviceCloneSelfCheck() error {
+	eng := engine.NewDeviceCloneTokenEngine(d.G())
+	m := libkb.NewMetaContextBackground(d.G())
+
+	beforeClones := eng.ClonesCount(m)
+	if err := engine.RunEngine2(m, eng); err != nil {
+		return err
+	}
+	newClones := eng.ClonesCount(m) - beforeClones
```

Comment:
Using an Engine this way is unusual. In general an Engine is a glorified function so its lifecycle is construct, Run, and optionally pull more results off it with methods. Can you get this information directly through `G.Env.GetConfig().GetDeviceCloneToken(un).Clones`?

Reply:
Agreed that it seems possible to put all code for this in one place!